pkg/frontend/admin_platformworkloadidentityroleset_put.go

```diff
@@ -0,0 +1,93 @@
+
+import (
+	"encoding/json"
+	"net/http"
+	"path/filepath"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/api/admin"
+	"github.com/Azure/ARO-RP/pkg/frontend/middleware"
+)
+
+func (f *frontend) putAdminPlatformWorkloadIdentityRoleSet(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	log := ctx.Value(middleware.ContextKeyLog).(*logrus.Entry)
+	r.URL.Path = filepath.Dir(r.URL.Path)
+
+	converter := f.apis[admin.APIVersion].PlatformWorkloadIdentityRoleSetConverter
+	staticValidator := f.apis[admin.APIVersion].PlatformWorkloadIdentityRoleSetStaticValidator
+
+	body := r.Context().Value(middleware.ContextKeyBody).([]byte)
+	if len(body) == 0 || !json.Valid(body) {
+		api.WriteError(w, http.StatusBadRequest, api.CloudErrorCodeInvalidRequestContent, "", "The request content was invalid and could not be deserialized.")
+		return
+	}
+
+	var ext *admin.PlatformWorkloadIdentityRoleSet
+	err := json.Unmarshal(body, &ext)
+	if err != nil {
+		api.WriteError(w, http.StatusBadRequest, api.CloudErrorCodeInvalidRequestContent, "", "The request content could not be deserialized: "+err.Error())
+		return
+	}
+
+	docs, err := f.dbPlatformWorkloadIdentityRoleSets.ListAll(ctx)
+	if err != nil {
+		api.WriteError(w, http.StatusInternalServerError, api.CloudErrorCodeInternalServerError, "", "Internal server error.")
+		return
+	}
+
+	var roleSetDoc *api.PlatformWorkloadIdentityRoleSetDocument
+	if docs != nil {
+		for _, doc := range docs.PlatformWorkloadIdentityRoleSetDocuments {
+			if doc.PlatformWorkloadIdentityRoleSet.Properties.OpenShiftVersion == ext.Properties.OpenShiftVersion {
+				roleSetDoc = doc
+				break
+			}
+		}
+	}
+
+	isCreate := roleSetDoc == nil
+	if isCreate {
+		err = staticValidator.Static(ext, nil)
+		roleSetDoc = &api.PlatformWorkloadIdentityRoleSetDocument{
+			ID:                              f.dbPlatformWorkloadIdentityRoleSets.NewUUID(),
+			PlatformWorkloadIdentityRoleSet: &api.PlatformWorkloadIdentityRoleSet{},
+		}
+	} else {
+		err = staticValidator.Static(ext, roleSetDoc.PlatformWorkloadIdentityRoleSet)
+	}
+	if err != nil {
+		adminReply(log, w, nil, []byte{}, err)
+		return
+	}
+
+	converter.ToInternal(ext, roleSetDoc.PlatformWorkloadIdentityRoleSet)
+
+	if isCreate {
+		roleSetDoc, err = f.dbPlatformWorkloadIdentityRoleSets.Create(ctx, roleSetDoc)
+		if err != nil {
+			api.WriteError(w, http.StatusInternalServerError, api.CloudErrorCodeInternalServerError, "", "Internal server error.")
+			return
+		}
+	} else {
+		roleSetDoc, err = f.dbPlatformWorkloadIdentityRoleSets.Update(ctx, roleSetDoc)
+		if err != nil {
+			api.WriteError(w, http.StatusInternalServerError, api.CloudErrorCodeInternalServerError, "", "Internal server error.")
+			return
+		}
+	}
+
+	b, err := json.MarshalIndent(converter.ToExternal(roleSetDoc.PlatformWorkloadIdentityRoleSet), "", "    ")
+	if err == nil {
+		if isCreate {
+			err = statusCodeError(http.StatusCreated)
```

Comment:
nit: idk if there's any way to make this read better, but this line of code hurts my brain - we are returning a 201, do we gotta have `statusCodeError` to set that?

Reply:
I took this from existing code, and it's done this way in difference places throughout the frontend package, so I'd like to leave this one as-is to avoid scope creep if that's okay.